pkg/reconcilermanager/controllers/reconciler_base.go

```diff
@@ -329,13 +436,18 @@ func keepCurrentContainerResources(declared, current *appsv1.Deployment) bool {
 // deployment returns the deployment from the server
 func (r *reconcilerBase) deployment(ctx context.Context, dRef client.ObjectKey) (*appsv1.Deployment, error) {
 	depObj := &appsv1.Deployment{}
-	if err := r.client.Get(ctx, dRef, depObj); err != nil {
+	depUnObj, err := r.dynamicClient.Resource(kinds.DeploymentResource()).Namespace(dRef.Namespace).Get(ctx, dRef.Name, metav1.GetOptions{})
+
+	if err != nil {
 		if apierrors.IsNotFound(err) {
 			return nil, errors.Errorf(
 				"Deployment %s not found in namespace: %s.", dRef.Name, dRef.Namespace)
 		}
 		return nil, errors.Wrapf(err, "error while retrieving deployment")
 	}
+	if err = runtime.DefaultUnstructuredConverter.FromUnstructured(depUnObj.Object, depObj); err != nil {
```

Comment:
Can `kinds.ToTypedObject` be used instead? I've been trying to route conversion through those helper utils, rather than using the converter directly.